plugins/flutter_plugins/shared_preferences_fde/macos/Classes/SharedPreferencesPlugin.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import FlutterMacOS
+import Foundation
+
+public class SharedPreferencesPlugin: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/shared_preferences",
+      binaryMessenger: registrar.messenger)
+    let instance = SharedPreferencesPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    let method = call.method
+    if method == "getAll" {
+      result(getAllPrefs())
+    } else if method == "setBool" ||
+              method == "setInt" ||
+              method == "setInt" ||
+              method == "setDouble" ||
+              method == "setString" ||
+              method == "setStringList" {
+      let arguments = call.arguments as! [String : Any]
```

Comment:
Should these be extracted with "if let"s? We've found that if a plugin crashes, the whole FDE crashes as well.

Reply:
These arguments are constructed internally to the Dart plugin, not user input, so failing this check would be a serious bug in the plugin itself. We should decide how to handle that generally as well.

Reply:
Sounds good, thanks!